### PR TITLE
Avoid list mutation in pickle_loads

### PR DESCRIPTION
**Pull request by mrocklin**
Fixes https://github.com/dask/distributed/issues/4625

- [ ] Closes #xxxx
- [ ] Tests added / passed
- [ ] Passes `black distributed` / `flake8 distributed`

### CODE REVIEW

File: distributed/protocol/serialize.py [hunk 1]
```diff
@@ -69,19 +69,21 @@ def pickle_dumps(x, context=None):
...
 def pickle_loads(header, frames):
     x, buffers = frames[0], frames[1:]
-    writeable = header["writeable"]
-    for i in range(len(buffers)):
-        mv = memoryview(buffers[i])
-        if writeable[i] == mv.readonly:
+
+    new = []
+    memoryviews = map(memoryview, buffers)
+    for writeable, mv in zip(header["writeable"], memoryviews):
```
Comment (jakirkham) [inline distributed/protocol/serialize.py:75]: This would be an issue if `header["writeable"]` is `()`, which came up in another issue

Comment (mrocklin) [reply distributed/protocol/serialize.py:75]: This PR is focused on removing the list mutation.  I recommend that we handle the `not header["writeable"]` issue separately if that's ok.

**Comment by bnaul** [comment]
can confirm this fixes the specific regression affecting us (no idea about the broader impact though)

**Comment by mrocklin** [comment]
Merging on passed tests.

@jrbourbeau can we schedule a bugfix release for this Friday at the latest?  I'll resolve the writeable thing too.

**Comment by jakirkham** [comment]
Yeah I was trying to prod CI last night, but I guess it still wasn't passing unfortunately (not that I think the failure is related)

**Pull request merged**